### PR TITLE
feat(resourcename): add ValidatePattern

### PR DESCRIPTION
**Pull request by odsod**
Disambiguates between resource name validation and pattern validation.

### CODE REVIEW

File: resourcename/validate.go [hunk 1]
```diff
@@ -30,3 +34,51 @@ func Validate(name string) error {
 	}
 	return nil
 }
+
+// ValidatePattern that a resource name pattern conforms to the restrictions outlined in AIP-122,
```
Comment (ericwenn) [inline resourcename/validate.go:38]: The comment for this function seems wrong. It should not mention names, but patterns.

Comment (odsod) [reply resourcename/validate.go:38]: It does say `resource name pattern` - what do you think should be updated?

Comment (ericwenn) [reply resourcename/validate.go:38]: Missed that. But it still only mentions DNS names, while it actually checks many more things. Either list everything, or include nothing imo.

**Review by ericwenn — APPROVED** [review]

File: resourcename/validate.go (hunk 1 above)
Comment (odsod) [reply resourcename/validate.go:38]: Will remove!

**Pull request merged**